libraries/AC_CustomControl/AC_CustomControl.cpp

```diff
@@ -0,0 +1,175 @@
+#include <AP_HAL/AP_HAL.h>
+
+#include "AC_CustomControl.h"
+
+#if CUSTOMCONTROL_ENABLED
+
+#include "AC_CustomControl_Backend.h"
+#include "AC_CustomControl_Empty.h"
+#include "AC_CustomControl_PID.h"
+
+// table of user settable parameters
+const AP_Param::GroupInfo AC_CustomControl::var_info[] = {
+    // @Param: CONT_TYPE
+    // @DisplayName: Attitude control type
+    // @Description: Attitude control type to be used
+    // @Values: 0:None, 1:Empty, 2:PID
+    // @RebootRequired: True
+    // @User: Advanced
+    AP_GROUPINFO_FLAGS("CONT_TYPE", 1, AC_CustomControl, _controller_type, 0, AP_PARAM_FLAG_ENABLE),
+
+    // @Param: CUST_CNT_MSK
+    // @DisplayName: Custom Controller bitmask
+    // @Description: Custom Controller bitmask to chose which axis to run
+    // @Bitmask: 0:Roll, 1:Pitch, 2:Yaw
+    // @User: Advanced
+    AP_GROUPINFO("AXIS_MASK", 2, AC_CustomControl, _custom_controller_mask, 0),
+
+    // parameters for empty controller
+    AP_SUBGROUPVARPTR(_backend[0], "1_", 6, AC_CustomControl, _backend_var_info[0]),
+
+    // parameters for PID controller
+    AP_SUBGROUPVARPTR(_backend[1], "2_", 7, AC_CustomControl, _backend_var_info[1]),
```

Comment:
Maybe better to remove the unserscore in copter and add it back where it is needed. That would allow this to be `CSC2_RAT_PIT_I` rather than `CSC_2_RAT_PIT_I` for example.

Reply:
Yeah, were getting truncated params in the PID backend, so we need to save some characters in the prefix somewhere.

Reply:
I didn't realize we were hitting the 16 char limit on parameter names. Changing CSC to CC.